activerecord/test/cases/associations/required_test.rb

```diff
@@ -22,14 +22,19 @@ class Child < ActiveRecord::Base
     @connection.drop_table "children", if_exists: true
   end
 
-  test "belongs_to associations are not required by default" do
+  test "belongs_to associations can be optional by default" do
+    original_value = ActiveRecord::Base.belongs_to_required_by_default
+    ActiveRecord::Base.belongs_to_required_by_default = false
+
     model = subclass_of(Child) do
       belongs_to :parent, inverse_of: false,
         class_name: "RequiredAssociationsTest::Parent"
     end
 
     assert model.new.save
     assert model.new(parent: Parent.new).save
+
+    ActiveRecord::Base.belongs_to_required_by_default = original_value
```

Comment:
It's not possible use `ensure` with the `test` DSL...

Would be better use a `begin ... ensure` block or change the tests to `def test_name_test` as used on other tests? If the last... do that for only those touched tests or the entire file?